src/Storage/ClaimController.ts

```diff
@@ -234,6 +235,32 @@ export class ClaimController {
     }
   }
 
+  private setDownloadSuccessTime = async ({
+    entry,
+    downloadSuccessTime = new Date().getTime(),
+    ...rest
+  }: {
+    entry: Entry
+    downloadSuccessTime?: number
+  }) => {
+    const logger = this.logger.child({ method: 'setDownloadSuccessTime' })
+    logger.trace('setting download success time')
+
+    await this.collection.updateOne(
+      {
+        _id: entry._id,
+      },
+      { $set: { downloadSuccessTime } }
+    )
+
+    logger.trace('finished setting download success time')
+
+    return {
+      entry,
+      ...rest,
+    }
+  }
+
```

Comment:
This function is doing two things: adding a downloadSuccessTime prop if it doesn't exist in it's argument AND updating the downloadSuccessTime in the database (whether it was missing in 'entry' or not).

That first part is a piece of conditional logic that can be extracted and unit tested. 
Something like this (not completely sure about the mongo call):

```javascript
  const updateWithCurrentTime = obj => Promise.resolve({ downloadSuccessTime: new Date().getTime(), ...obj})

  const updateDatabase = async (entry) => {
    await this.collection.updateOne(
      {
        _id: entry._id,
      },
      { $set: { downloadSuccessTime: entry.downloadSuccessTime } }
    )

    return entry
  }

  public setDownloadSuccessTime = pipeM(updateWithCurrentTime, updateDatabase)
```

The `updateWithCurrentTime` function would have 2 unit tests to handle the conditional:
  - given an object without a dowloadSuccessTime prop
    should return an object with the prop added
  - given an object with a downloadSuccessTiome prop
    should return an the same object.

`updateWithCurrentTime` could even be generalized further by taking the prop name if it would be useful in other areas of the codebase.

Reply:
I agree with @warrenv if we can use function composing cool

Reply:
@warrenv and I already discussed this via zoom. What he proposed only works if I am modifying the pass-through data which I am not anymore. I removed that logic from this PR.